### PR TITLE
029 Arima Component

### LINKED ISSUE #29: Time Series prediction using ARIMA
## User Story

1. As a RTDIP user
2. I want to predict the future development of periodical, time series data, given past values
3. So that I can plan ahead

## Additional context

- Similar to #28 
- A library that could be used: https://www.statsmodels.org/stable/generated/statsmodels.tsa.arima.model.ARIMA.html

## Acceptance Criteria

- Component input
  - past data
  - the name of the column with the data that should be predicted
  - point in time in the future, up to which the data will be generated (or the number of data points to predict in the same interval)
- Component output
  - past data + future predictions 
  - (for all other columns except the predicted one, the values can be empty)
- The component will analyze past time series, and apply the ARIMA model
- The component will generate new data up to a given point of time in the future, using the ARIMA model

## Definition of Done

- Test cases have been created and are running successfully
- Documentation for the new component was added
- Github Actions are running without errors

### PR DESCRIPTION
**Pull request by Timm638**

### CODE REVIEW

File: tests/sdk/python/rtdip_sdk/pipelines/data_wranglers/spark/data_quality/test_arima.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        number_of_data_points_to_analyze=h_a_l,
+        number_of_data_points_to_predict=h_a_l,
+        order=(3, 0, 0),
+        seasonal_order=(3, 0, 0, 62),
+    )
+    forecasted_df = arima_comp.filter()
+
+    assert isinstance(forecasted_df, DataFrame)
+
+    assert forecasted_df.columns == forecasted_df.columns
```
Comment (FelipeTrost) [inline tests/sdk/python/rtdip_sdk/pipelines/data_wranglers/spark/data_quality/test_arima.py]: isn't this just a comparison between the same columns?

File: tests/sdk/python/rtdip_sdk/pipelines/data_wranglers/spark/data_quality/test_arima.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    # Workaround needed for PySpark versions <3.4
+    if not hasattr(learn_df, "iteritems"):
+        learn_df.iteritems = learn_df.items
+
+    input_df = spark_session.createDataFrame(
+        learn_df,
+        ["Value", "Value2", "index"],
+    )
+
+    # input_df.show(n=50)
```
Comment (FelipeTrost) [inline tests/sdk/python/rtdip_sdk/pipelines/data_wranglers/spark/data_quality/test_arima.py]: left over comment

**Review by FelipeTrost — COMMENTED** [review]
I doesn't seem like the test is checking the output of the component

File: tests/sdk/python/rtdip_sdk/pipelines/data_wranglers/spark/data_quality/test_arima.py (hunk 1 above)
Comment (Timm638) [reply tests/sdk/python/rtdip_sdk/pipelines/data_wranglers/spark/data_quality/test_arima.py]: Fixed that typo

**Comment by Timm638** [comment]
> I doesn't seem like the test is checking the output of the component

The scope of the test is to check if the integration of statsmodels is working in the RTDIP pipeline. It's not the goal of the test to check if the method provided by statsmodel is working correctly.

File: tests/sdk/python/rtdip_sdk/pipelines/data_wranglers/spark/data_quality/test_arima.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from abc import ABCMeta
+
+import numpy as np
+import pandas
```
Comment (chris-1187) [inline tests/sdk/python/rtdip_sdk/pipelines/data_wranglers/spark/data_quality/test_arima.py]: Is this a duplicate of the next line?

**Comment by FelipeTrost** [comment]
> st is to check if the integration of statsmodels is working in the RTDIP pipeline. It's not the goal of the test to check if the method provided by statsmodel is working correctly.

I would argue, that we should test that the output conforms to what we expect, because changes in the library you used or even a change of library on our side, could change the output, and components shouldn't change how they behave.

But still, I see your point, that the test wouldn't be actually testing something under our control. And I also think, it is safe to assume, that they tested their components :laughing:

So, if you take care of the pipeline errors and ping me, I will merge it immediatly.

**Comment by chris-1187** [comment]
LGTM

**Pull request merged**